src/logic/map_objects/tribes/soldiercontrol.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "logic/map_objects/tribes/soldiercontrol.h"
+
+#include <memory>
+
+#include "base/i18n.h"
+#include "base/string.h"
+#include "graphic/style_manager.h"
+#include "logic/map_objects/tribes/tribe_descr.h"
+
+namespace {
+
+std::string format_extra_soldiers(int has, int wants) {
```

Comment:
I guess these could or even should be `Quantity`-es, but that shouldn't really matter for casting, should it?